Improve changefeed resolved message description (#17120)
cockroachdb/docs: src/current/v23.1/changefeed-for.md

## Required privileges

{{site.data.alerts.callout_info}}

> Starting in v22.2, CockroachDB introduces a new [system-level privilege model](security-reference/authorization.html#supported-privileges) that provides finer control over a user's privilege to work with the database, including creating and managing changefeeds.

Reviewer: ```suggestion
In v22.2 and above, CockroachDB introduces a new [system-level privilege model](security-reference/authorization.html#supported-privileges) that provides more fine-grained control over a user's privilege to work with the database, including the ability to create and manage changefeeds.
```
By "the database" do you mean the cluster?

The "including creating..." seems like a bit of a dangling modifier, since the main verb in the sentence is "introduces" and the sub-clause verb is "provides". I've rewritten it a bit. WDYT?

Author: Yeah that's fine, this isn't something that I wrote personally, so lgtm

> In v22.2 and above, CockroachDB introduces a new [system-level privilege model](security-reference/authorization.html#supported-privileges) that provides more fine-grained control over a user's privilege to work with the database, including the ability to create and manage changefeeds.

Reviewer: ```suggestion
In v22.2 and above, CockroachDB introduces a new [system-level privilege model](security-reference/authorization.html#supported-privileges) that provides more fine-grained control over a user's privileges to work with the cluster, including the ability to create and manage changefeeds.
```

There is continued support for the [legacy privilege model](#legacy-privilege-model) for changefeeds in v23.1, however it **will be removed** in a future release of CockroachDB. We recommend implementing the new privilege model that follows in this section for all changefeeds.
{{site.data.alerts.end}}

> {% include_cached new-in.html version="v23.1" %} Users require `SELECT` usage on a table to create a changefeed with `EXPERIMENTAL CHANGEFEED FOR`.

Reviewer: Do you mean something like "To create a changefeed with the `EXPERIMENTAL CHANGEFEED FOR` clause, a user must have the `SELECT` privilege on the changefeed's source tables." I'm confused about what "SELECT usage" means.

`mvcc_timestamp` | N/A |  Include the [MVCC](architecture/storage-layer.html#mvcc) timestamp for each emitted row in a changefeed. With the `mvcc_timestamp` option, each emitted row will always contain its MVCC timestamp, even during the changefeed's initial backfill.

> <a name="resolved-option"></a>`resolved` | [`INTERVAL`](interval.html) | Emits [resolved timestamp](changefeed-messages.html#resolved-messages) events for the changefeed. Resolved timestamp events do not emit until all ranges in the changefeed have progressed to a specific point in time. <br><br>Set an optional minimal duration between emitting resolved timestamps. Example: `resolved='10s'`. This option will **only** emit a resolved timestamp event if the timestamp has advanced and at least the optional duration has elapsed. If a duration is unspecified, all resolved timestamps are emitted as the high-water mark advances.<br><br>**Note:** If you require `resolved` message frequency under `30s`, then you **must** set the [`min_checkpoint_frequency`](#min-checkpoint-frequency) option to at least the desired `resolved` frequency. This is because `resolved` messages will not be emitted more frequently than `min_checkpoint_frequency`, but may be emitted less frequently.

Reviewer: ```suggestion
<a name="resolved-option"></a>`resolved` | [`INTERVAL`](interval.html) | Emits [resolved timestamp](changefeed-messages.html#resolved-messages) events for the changefeed. Resolved timestamp events are not emitted until all of the changefeed's ranges have progressed to the specified duration. <br><br>Example: `resolved='10s'`. If a duration is unspecified, all resolved timestamps are emitted as the high-water mark advances.<br><br>**Note:** If you set `resolved` lower than `30s`, then you **must** also set [`min_checkpoint_frequency`](#min-checkpoint-frequency) to at minimum the same value as `resolved`, because `resolved` messages may be emitted less frequently than `min_checkpoint_frequency`, but cannot be emitted more frequently.
```
Hopefully I haven't misunderstood this.

> <a name="resolved-option"></a>`resolved` | [`INTERVAL`](interval.html) | Emits [resolved timestamp](changefeed-messages.html#resolved-messages) events for the changefeed. Resolved timestamp events do not emit until all ranges in the changefeed have progressed to a specific point in time. <br><br>Set an optional minimal duration between emitting resolved timestamps. Example: `resolved='10s'`. This option will **only** emit a resolved timestamp event if the timestamp has advanced and at least the optional duration has elapsed. If a duration is unspecified, all resolved timestamps are emitted as the high-water mark advances.<br><br>**Note:** If you set `resolved` lower than `30s`, then you **must** also set the [`min_checkpoint_frequency`](#min-checkpoint-frequency) option to at minimum the same value as `resolved`, because `resolved` messages may be emitted less frequently than `min_checkpoint_frequency`, but cannot be emitted more frequently.<br><br>Refer to [Resolved messages](changefeed-messages.html#resolved-messages) for more detail.

Reviewer: ```suggestion
<a name="resolved-option"></a>`resolved` | [`INTERVAL`](interval.html) | Emits [resolved timestamp](changefeed-messages.html#resolved-messages) events for the changefeed. Resolved timestamp events do not emit until all ranges in the changefeed have progressed to a specific point in time. <br><br>Set an optional minimal duration between emitting resolved timestamps. Example: `resolved='10s'`. This option will emit a resolved timestamp event **only** if the timestamp has advanced and at least the optional duration has elapsed. If no duration is specified, all resolved timestamps are emitted as the high-water mark advances.<br><br>**Note:** If you set `resolved` lower than `30s`, then you **must** also set the [`min_checkpoint_frequency`](#min-checkpoint-frequency) option to at minimum the same value as `resolved`, because `resolved` messages may be emitted less frequently than `min_checkpoint_frequency`, but cannot be emitted more frequently.<br><br>Refer to [Resolved messages](changefeed-messages.html#resolved-messages) for more detail.
```
I can't remember what the grammar rule is called, but the "only" should be as near as possible to the condition. :)

> <a name="resolved-option"></a>`resolved` | [`INTERVAL`](interval.html) | Emits [resolved timestamp](changefeed-messages.html#resolved-messages) events for the changefeed. Resolved timestamp events do not emit until all ranges in the changefeed have progressed to a specific point in time. <br><br>Set an optional minimal duration between emitting resolved timestamps. Example: `resolved='10s'`. This option will **only** emit a resolved timestamp event if the timestamp has advanced and at least the optional duration has elapsed. If a duration is unspecified, all resolved timestamps are emitted as the high-water mark advances.<br><br>**Note:** If you set `resolved` lower than `30s`, then you **must** also set the [`min_checkpoint_frequency`](#min-checkpoint-frequency) option to at minimum the same value as `resolved`, because `resolved` messages may be emitted less frequently than `min_checkpoint_frequency`, but cannot be emitted more frequently.<br><br>Refer to [Resolved messages](changefeed-messages.html#resolved-messages) for more detail.

Reviewer: In changefeed-messages.md you use `RESOLVED` but here you use `resolved`. Should they be the same or is that intentional?

Author: should be lowercased for consistency!
